tests/unit/tf/examples/test_usecase_incremental_training_layer_freezing.py

```diff
@@ -14,6 +15,10 @@
     execute=False,
 )
 @pytest.mark.notebook
+@pytest.mark.skipif(
+    version.parse(tf.__version__) < version.parse("2.9.0"),
+    reason="tf.keras.optimizers.legacy is not available in TF <= 2.8",
+)
```

Comment:
Since we need to use `legacy` optimizers in MultiOptimizer, I updated the incremental training notebook to use `tf.keras.optimizers.legacy` optimizers. And since this namespace was not available in TF 2.8, I opted to skip testing for TF 2.8, effectively dropping support for TF 2.8 for this notebook. Open questions: Is there a better way to handle this? It's likely not feasible to support all previous versions forever. Do we need an SLA or something for when to drop support for old dependency versions?

Reply:
Maybe the last 4 releases of TensorFlow or something would be reasonable. Might depend a bit on the level of complexity in supporting  each one